services/translate.cfc

```diff
@@ -0,0 +1,58 @@
+component {
+
+  variables.service = 'translate';
+
+  public any function init(
+      required any api,
+      required struct settings
+  ) {
+      variables.api = arguments.api;
+      return this;
+  }
+
+  public any function translate(required string text, string sourceLang='es', string targetLang='en' ){
+    var payload = {};
+    payload[ 'SourceLanguageCode' ] = sourceLang;
+    payload[ 'TargetLanguageCode'] = targetLang;
+    payload[ 'Text' ] = text;
+
+    var apiResponse = apiCall( payload = serializeJSON(payload) );
+    if ( apiResponse.statusCode == 200 ) {
+      apiResponse[ 'data' ] = deserializeJSON(apiResponse.rawData).TranslatedText;
+    }
+    return apiResponse;
+  }
+
+  public string function getHost(
+      required string region
+  ) {
+      return variables.service & '.' & region & '.amazonaws.com';
+  }
+
+  private any function apiCall(
+      string httpMethod = 'POST',
+      string path = '/',
+      struct queryParams = { },
+      struct headers = { },
+      any payload = ''
+  ) {
+    var requestSettings = api.resolveRequestSettings( argumentCollection = arguments );
```

Comment:
This should be in the `translate()` method. What I am trying to do is allow for any method to be passed AWS credentials directly (or an AWS region), and then have those settings used for that request. So `requestSettings` are parsed in each public method, and then passed to the `apiCall()` method.